### PR TITLE
[#400] Fix complete-multipart-upload

### LINKED ISSUE #400: Make sure complete-multipart-upload works correctly
It seems after successful completing multipart upload some system information is still preserved

```
denis@denis-nspcc:~$ aws s3api complete-multipart-upload --endpoint http://localhost:8084 --bucket new-bucket --key multipart-key2  --upload-id 5b4edd92-9aa3-4eff-8500-dd89fda59bbc --multipart-upload file://parts.json
{
    "Bucket": "new-bucket",
    "Key": "multipart-key2",
    "ETag": "5412930ac75fd19871dfe9084435e5dabb3873c7108c0c9d3273a35f20efbcac"
}

denis@denis-nspcc:~$ aws s3api list-parts --endpoint http://localhost:8084 --bucket new-bucket --key multipart-key2  --upload-id 5b4edd92-9aa3-4eff-8500-dd89fda59bbc
{
    "Initiator": {
        "ID": "NbUgTSFvPmsRxmGeWpuuGeJUoRoi6PErcM",
        "DisplayName": "NbUgTSFvPmsRxmGeWpuuGeJUoRoi6PErcM"
    },
    "Owner": {
        "DisplayName": "NbUgTSFvPmsRxmGeWpuuGeJUoRoi6PErcM",
        "ID": "NbUgTSFvPmsRxmGeWpuuGeJUoRoi6PErcM"
    },
    "StorageClass": null
}

denis@denis-nspcc:~$ aws s3api list-multipart-uploads --endpoint http://localhost:8084 --bucket new-bucket 
{
    "Uploads": [
        {
            "UploadId": "5b4edd92-9aa3-4eff-8500-dd89fda59bbc",
            "Key": "multipart-key2",
            "Initiated": "2022-04-11T14:40:16+00:00",
            "Owner": {
                "DisplayName": "NbUgTSFvPmsRxmGeWpuuGeJUoRoi6PErcM",
                "ID": "NbUgTSFvPmsRxmGeWpuuGeJUoRoi6PErcM"
            },
            "Initiator": {
                "ID": "NbUgTSFvPmsRxmGeWpuuGeJUoRoi6PErcM",
                "DisplayName": "NbUgTSFvPmsRxmGeWpuuGeJUoRoi6PErcM"
            }
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by masterSplinter01**
Moved deletion of parts after putting a final object, acl and tags.

Closes #400

### CODE REVIEW

File: api/handler/multipart_upload.go [hunk 1]
```diff
@@ -436,18 +436,9 @@ func (h *handler) CompleteMultipartUploadHandler(w http.ResponseWriter, r *http.
...
-	}
-
-	p := &layer.DeleteObjectParams{
-		BktInfo:     bktInfo,
-		BktSettings: bktSettings,
-		Objects:     []*layer.VersionedObject{{Name: initPart.Name}},
-	}
-	if _, err = h.obj.DeleteObjects(r.Context(), p); err != nil {
-		h.logAndSendError(w, "could not delete init file of multipart upload", reqInfo, err, additional...)
+	if err = h.obj.AbortMultipartUpload(r.Context(), uploadInfo); err != nil {
```
Comment (KirillovDenis) [inline api/handler/multipart_upload.go:439]: It's a little confused to see `AbortMultipartUpload` after `CompleteMultipartUpload`. In addition it leads to overhead to get all parts again. I think we can just use `DeleteSystemObject` to delete `initPart` here.

File: api/layer/multipart_upload.go [hunk 2]
```diff
@@ -500,18 +505,29 @@ func (n *layer) getUploadParts(ctx context.Context, p *UploadInfoParams) (map[in
 			continue
 		}
 		info := objInfoFromMeta(p.Bkt, meta)
+		// skip objects which are completed by "complete-multipart-upload" because they have "s3-Upload-Id" attribute
+		if !isSystem(info) {
+			continue
+		}
```
Comment (KirillovDenis) [inline api/layer/multipart_upload.go:511]: Could you explain this? As I understand all parts are system now so they will all be skipped.

File: api/layer/multipart_upload.go [hunk 3]
```diff
@@ -500,18 +505,29 @@ func (n *layer) getUploadParts(ctx context.Context, p *UploadInfoParams) (map[in
...
+
+	if len(res) == 0 {
+		return nil, errors.GetAPIError(errors.ErrNoSuchUpload)
 	}
 
 	return res, nil
 }
 
-func createUploadPartName(uploadID, key string, partNumber int) string {
+func UploadPartName(uploadID, key string, partNumber int) string {
```
Comment (KirillovDenis) [inline api/layer/multipart_upload.go:530]: I suppose we should add comment and/or change name to more clear (e.g.  `FormUploadPartName`).

File: api/layer/multipart_upload.go (hunk 2 above)
Comment (masterSplinter01) [reply api/layer/multipart_upload.go:511]: But we skip common files not system. We check **!** isSystem()

File: api/data/info.go [hunk 4]
```diff
@@ -14,6 +14,7 @@ const (
 	bktSettingsObject                  = ".s3-settings"
 	bktCORSConfigurationObject         = ".s3-cors"
 	bktNotificationConfigurationObject = ".s3-notifications"
+	ObjectSystemAttributeName          = "S3-System-name"
```
[the remark this answers is not in the document]
Comment (masterSplinter01) [reply api/data/info.go:17]: Oh, while writing the code it was being used in `data` package but in final version it's used only in `layer` package.  Will get back it to `layer`.

File: api/layer/versioning.go [hunk 5]
```diff
@@ -45,7 +45,7 @@ const (
 	DelMarkFullObject      = "*"
 
 	unversionedObjectVersionID    = "null"
-	objectSystemAttributeName     = "S3-System-name"
+	ObjectSystemAttributeName     = "S3-System-name"
```
Comment (KirillovDenis) [inline api/layer/versioning.go:48]: Can we not change this name at all?

**Review by KirillovDenis — APPROVED** [review]

**Review by alexvanin — APPROVED** [review]

**Pull request merged**